pkg/prometheus/agent/common.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2023 The prometheus-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheusagent
+
+import (
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/prometheus-operator/prometheus-operator/pkg/operator"
+	prompkg "github.com/prometheus-operator/prometheus-operator/pkg/prometheus"
+)
+
+func buildImagePath(
+	cpf monitoringv1.CommonPrometheusFields,
+	c *prompkg.Config,
+) (string, error) {
+	pImagePath, err := operator.BuildImagePath(
+		operator.StringPtrValOrDefault(cpf.Image, ""),
+		operator.StringValOrDefault("", c.PrometheusDefaultBaseImage),
+		operator.StringValOrDefault(cpf.Version, operator.DefaultPrometheusVersion),
+		"",
+		"",
+	)
+
+	if err != nil {
+		return "", err
+	}
+
+	return pImagePath, nil
+}
```

Comment:
Yeah one goal of the refactoring is to make reviewing #6652 easier. As @simonpasquier said in the last meeting, we need to refactor so that we won't need to go back to statefulset code to check if the duplicated daemonset code is correct or not.

If there are no hard opinions, I'll keep this refactoring. Without it you might need to go back to statefulset code to check if `operator.BuildImagePath()` is correctly called in daemonset or not, i.e., if the arguments are correctly set and the returned values are correctly handled.